src/platform/plugins/shared/visualizations/public/utils/saved_visualize_utils.ts

```diff
@@ -359,20 +359,27 @@ export async function saveVisualization(
   }
 
   try {
-    await checkForDuplicateTitle(
-      savedObject,
-      copyOnSave,
-      isTitleDuplicateConfirmed,
-      onTitleDuplicate,
-      services
-    );
-    const createOpt = {
-      migrationVersion: savedObject.migrationVersion,
-      references: extractedRefs.references,
-    };
+    if (onTitleDuplicate) {
+      // Only checks dups if onTitleDuplicate is passed.
+      // The save action is done from multiple places. In some cases like the embeddable,
+      // the title dup check is done before save is called and in other places we need to check.
+      await checkForDuplicateTitle(
+        savedObject,
+        copyOnSave,
+        isTitleDuplicateConfirmed,
+        onTitleDuplicate
+      );
+    }
```

Comment:
Only check the duplicate title on save when `onTitleDuplicate` is defined. Avoids calling when saving from the dashboard.